pkg/logs/internal/decoder/single_line_handler.go

```diff
@@ -36,31 +37,39 @@ func (h *SingleLineHandler) flush() {
 	// do nothing
 }
 
-// process transforms a raw line into a structured line,
-// it guarantees that the content of the line won't exceed
-// the limit and that the length of the line is properly tracked
-// so that the agent restarts tailing from the right place.
+// process transforms a raw line into a structured line, it guarantees that the
+// content of the line won't exceed the limit and that the length of the line is
+// properly tracked so that the agent restarts tailing from the right place.
 func (h *SingleLineHandler) process(message *message.Message) {
 	isTruncated := h.shouldTruncate
 	h.shouldTruncate = false
 
 	message.Content = bytes.TrimSpace(message.Content)
 
+	h.buffer = h.buffer[:0] // reset, avoid polluting different process calls
+
 	if isTruncated {
 		// the previous line has been truncated because it was too long,
-		// the new line is just a remainder,
-		// adding the truncated flag at the beginning of the content
-		message.Content = append(truncatedFlag, message.Content...)
+		// the new line is just a remainder, adding the truncated flag
+		// at the beginning of the content
+		h.buffer = append(h.buffer, truncatedFlag...)
+		h.buffer = append(h.buffer, message.Content...)
+		message.Content = h.buffer
```

Comment:
It looks like the previous `Message` will be made invalid each time `SingleLineHandler.process` is called. Is that okay?

Reply:
Hmm, by invalid do you mean truncated? I think, no. That strikes me as a bug.

Reply:
Yeah, it looks like each invocation will send out a pointer to the same buffer. That'll be a problem if they're held onto or called in parallel. I'm not 100% sure on that though.